e2e-tests/IIoTPlatform-E2E-Tests/deploy/IoTHubPublisherDeployment.cs

```diff
@@ -0,0 +1,205 @@
+﻿// ------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All rights reserved.
+//  Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
+// ------------------------------------------------------------
+
+namespace IIoTPlatform_E2E_Tests.TestExtensions {
+    using IIoTPlatform_E2E_Tests.deploy;
+    using Microsoft.Azure.Devices.Common.Exceptions;
+    using System;
+    using System.Collections.Generic;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Newtonsoft.Json;
+
+    public class IoTHubPublisherDeployment {
+
+        public static readonly string TargetConditionStandalone =
+            $"(tags.__type__ = 'iiotedge' AND tags.unmanage = 'false')";
+
+        /// <summary>
+        /// Create deployer
+        /// </summary>
+        /// <param name="context"></param>
+        public IoTHubPublisherDeployment(IIoTPlatformTestContext context) {
+            _context = context ?? throw new ArgumentNullException(nameof(context));
+        }
+
+        /// <summary>
+        /// Create a new layered deployment or update an existing one
+        /// </summary>
+        public async Task CreateOrUpdateLayeredDeploymentAsync() {
+            await CreateOrUpdateConfigurationAsync(new ConfigurationModel {
+                Id = "__default-opcpublisher-standalone",
+                Content = new ConfigurationContentModel {
+                    ModulesContent = CreateLayeredDeployment(true, true)
+                },
+                SchemaVersion = kDefaultSchemaVersion,
+                TargetCondition = TargetConditionStandalone +
+                    " AND tags.os = 'Linux'",
+                Priority = 1
+            }, true, new CancellationToken());
+        }
+
+        /// <summary>
+        /// Get base edge configuration
+        /// </summary>
+        /// <returns></returns>
+        private IDictionary<string, IDictionary<string, object>> CreateLayeredDeployment(bool isLinux, bool isStandalone) {
```

Comment:
>bool isStandalone [](start = 103, length = 17)

Please remove, in our use case we are only linux and only standalone

Reply:
removed